### PR TITLE
Update pdf_generator.ex

### PR DESCRIPTION
**Pull request by edipox**
Add the `html_file` as part of to the returned value. 

Use case: someone wants to remove the generated HTML temporary file

### CODE REVIEW

**Comment by gutschilla** [comment]
@edipox Thanks for your PR! Usually one shouldn't care about temporary files (your OS should do) so explicit deleting isn't usually necessary. 

On the other hand there's no real reason not to be able to delete that file but I'd rather see an option to the call than changing it's return value signature from a 2-element tuple to a 3-element tuple as this would surely break someone's code (at least mine) when pattern-matching the result of `generate` as in

```
{:ok, file} = PdfGenerator.generate(...)
```
So I'd rather suggest an option `delete_temporay: :html` that does just that. 

Tell what you think

**Comment by edipox** [comment]
You're right about the code breaking impact. 
I have a specific use case: concurrent PDF generation where files can't be left for OS cleanup (disk space concerns) or batch-deleted (race conditions with other requests reading files). 
The `delete_temporary: :html` option solves this elegantly.
Thank you!

**Comment by edipox** [comment]
@gutschilla I did what you suggested. Could you take a look?

File: lib/pdf_generator.ex [hunk 1]
```diff
@@ -133,6 +135,11 @@ defmodule PdfGenerator do
       executable, arguments, [in: "", out: :string, err: :string]
     )
 
+    case Keyword.get(options, :delete_temporary) do
```
Comment (gutschilla) [inline lib/pdf_generator.ex]: That could be rewritten as 
```
if Keyword.get(options, :delete_temporary) == :html do
   File.rm html_file
end
```
In my opinion, no need for a `nil -> nil` case

**Comment by gutschilla** [comment]
Thanks a lot for your patch and thanks as well for removing all these spaces at line endings. Would you please take a look at my small-ish line comment?

**Comment by edipox** [comment]
@gutschilla Done! 
The end line spaces were removed by this Atom package: https://github.com/atom/whitespace each time you save a file the package remove spaces automatically

**Comment by gutschilla** [comment]
Beautiful! Thanks a lot for your contribution!

**Pull request merged**